pandas/core/generic.py

```diff
@@ -9582,6 +9582,8 @@ def _where(
         cond = common.apply_if_callable(cond, self)
         if isinstance(cond, NDFrame):
             cond, _ = cond.align(self, join="right", broadcast_axis=1, copy=False)
+            # CoW: Make sure reference goes out of scope
+            _ = None
```

Comment:
would it be easier to simply do `cond = cond.align(self, join="right", broadcast_axis=1, copy=False)[0]` to avoid having a reference in the first place?

Reply:
Yikes, yes definitely. Thx